bibtex-actions.el

```diff
@@ -97,6 +97,29 @@ manager like Zotero or JabRef."
   :group 'bibtex-actions
   :type '(repeat function))
 
+(defcustom bibtex-actions-initial-inputs
+  '((pdf    . "has:pdf")
+    (note   . "has:note")
+    (link   . "has:link")
+    (source . "has:link\\|has:pdf"))
```

Comment:
```suggestion
  '((pdf    . "has:pdf ")
    (note   . "has:note ")
    (link   . "has:link ")
    (source . "has:link\\|has:pdf "))
```

Right? So point is separated from initial input with a space?

Reply:
Space is added in `bibtex-actions-read`, which see.

As to the use case, I thought the idea is to just make it customizable anyways. It's Emacs, after all. :smile: 

For me, at least, there _is_ actually a good use case: I use `bibtex-action-open-notes` with the fallback mechanism provided by `bibtex-completion`, which creates a note file if there is none. So the initial input is useless, since for me, `bibtex-action-open-notes` is supposed to open _or_ create a note with one command (which is very handy, btw). Another use case would be the ability to define what "source" actually means (as discussed in #121 ).

But TBH, I cannot imagine many more use cases right now. :shrug: So it's up to the maintainer!

Reply:
> Space is added in `bibtex-actions-read`, which see.

Oops! I just hid the comment then.

> As to the use case, I thought the idea is to just make it customizable anyways. It's Emacs, after all. 

Well, no; customization for the sake of it would be overkill.

But when there's need, that's another matter.

> For me, at least, there _is_ actually a good use case: I use `bibtex-action-open-notes` with the fallback mechanism provided by `bibtex-completion`, which creates a note file if there is none. 

Ooh, that's actually one I care about too; I had totally forgot about that!

This does raise a larger design question, which is what @apc and I were going back-and-forth on in #118.

We settled on the initial-input approach, because the goal was filtering, but to make it easy for the user to effectively turn off or modify the filtering mid-stream; in this case to delete the initial-input.

This not only because of cases like this, but also because with embark, you may open your library with one action in mind, but change your mind, and so need access to a wider list of candidates.

_**So the big question: did we make the right decision? And if yes, is this the right addition to this?**_

I'm inclined to say yes on both counts, but curious what you and @apc think.

> So the initial input is useless, since for me, `bibtex-action-open-notes` is supposed to open _or_ create a note with one command (which is very handy, btw). Another use case would be the ability to define what "source" actually means (as discussed in #121 ).
> 
> But TBH, I cannot imagine many more use cases right now.  So it's up to the maintainer!

No, that's a good enough reason, per above.